data/ai/lua/ca_recruit_rushers.lua

```diff
@@ -944,6 +944,14 @@ function ca_recruit_rushers:evaluation(cfg, data, filter_own)
         end
     end
 
+    recruit_data.recuit_types = {}
+    for _,unit_type in ipairs(wesnoth.sides[wesnoth.current.side].recruit) do
+        recruit_data.recuit_types[unit_type] = true
```

Comment:
Missing the R in "recruit_types" here.

Reply:
Funny in that I had just noticed that myself.  And there was a more substantial problem also: the code setting up that table actually had to be pulled forward about 20 lines as it was sometimes needed earlier (a situation that had not happened in my previous tests).  Both is fixed now.